src/useGetPosts/index.js

```diff
@@ -67,22 +60,23 @@ import { apiClient } from '../utils';
 function useGetPosts( args = {}, clientId, postType ) {
 	const [ options, setOptions ] = useState( [] );
 	const [ query, setQuery ] = useState( '' );
-	const [ loading, setLoading ] = useToggle();
+	const [ loading, setLoading ] = useState( true );
```

Comment:
@kuserich I assume we can revert the initial state value to `false` as the indicator is set to be `true` later inside the effect hook.

Reply:
Personally, as a developer, I would find the `useToggle` more distinguishable than the state hook. It would immediately strike me as a primitive boolean without tracking the usage of the setter function inside the hook.

Reply:
Yes, I agree, I would also find `useToggle` more distinguishable in use cases like this. However, with `useToggle` we're unable to set the state in cases where we do not want to toggle (i.e. ensure it is `true`). The original default state `false` is only required because we need to toggle the state value immediately before querying the post data for all states of the component (first query vs subsequent queries). That is, we are mapping the state logic into a toggle state logic. In fact, the default loading state could be set to `true` and the toggle operation could be conditional instead. Consider the following example:

```JS
function useGetPosts( args = {}, clientId, postType ) {
    const [ options, setOptions ] = useState( [] );
    const [ query, setQuery ] = useState( '' );
    const [ loading, setLoading ] = useToggle( false );
    const toast = useToast();

    useDeepCompareEffect( () => {
        if ( ! loading ) {
            setLoading();
        }
        apiClient
            .get( `/wp/v2/${ postType }`, { per_page: -1, post_status: 'publish', ...args } )
            .then( ( data ) => {
                setOptions( selectOptions( data, { id: 'value', 'title.rendered': 'label' }, [] ) );
                setQuery( data );
                setLoading();
            } )
            .catch( () => {
                setQuery( [] );
                setLoading();
                toast( __( 'Error: Couldn’t retrieve posts via API.', 'sixa' ), 'error' );
            } );
    }, [ args, clientId, postType ] );

    return { isLoading: loading, postsOptions: options, postsQuery: query };
}
```

Notice the new condition

```JS
if ( ! loading ) {
    setLoading();
}
```

With this, we are always setting `loading` to `true` before the query (and consequently always setting `loading` to `false` afterwards). This would be equivalent to the `setState` behavior. However, the condition, I believe, bloats the hook unnecessarily. We would be able to provide a custom toggle function that supports this feature. However, this might add additional complexity and might bloat the hook even further.

Reply:
@kuserich Thanks for clarifying 👍 Let's move forward with the state hook as it would not require the extra conditional statement added to the hook. 🙂